src/header.rs

```diff
@@ -0,0 +1,180 @@
+use ::Mp3Error;
+use tables::*;
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Version {
+    Mpeg2_5,
+    Reserved,
+    Mpeg2,
+    Mpeg1,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Layer {
+    LayerI,
+    LayerII,
+    LayerIII,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Mode {
+    Stereo,
+    JointStereo,
+    DualChannel,
+    Mono,
+}
+
+/*
+|------|----------------|------------------------------|
+|      | Layer I and II |          Layer III           |
+|------|----------------|------------------|-----------|
+| bits |  Layer I & II  | Intensity Stereo | MS Stereo |
+|------|----------------|------------------|-----------|
+|  00  | bands 4 to 31  |        Off       |    Off    |
+|  01  | bands 8 to 31  |        On        |    Off    |
+|  10  | bands 12 to 31 |        Off       |    On     |
+|  11  | bands 16 to 31 |        On        |    On     |
+|------|----------------|------------------|-----------|
+*/
+#[derive(Debug, Clone, PartialEq)]
+pub enum ModeExtension {
+    Bands(u8),
+    Stereo(bool, bool),
+}
+
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Emphasis {
+    None,
+    Ms50_15,  // 50/15 ms
+    Reserved,
+    CcittJ17,  // CCITT J.17
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct FrameHeader {
+    version: Version,
+    layer: Layer,
+    protection: bool,
+    pub bitrate: u16, // kbps
+    pub sampling_rate: u16, // Hz
+    pub padding: bool,
+    private: bool,
+    mode: Mode,
+    mode_extension: ModeExtension,
+    copyright: bool,
+    original: bool,
+    emphasis: Emphasis,
+}
+
+
+pub fn parse_frame_header(data: &[u8]) -> Result<FrameHeader, Mp3Error> {
+    let header = &data[..4];
+
+    // Sync word check
+    if (header[0] != 255_u8) && (header[1] < 0b11100000u8) {
+        return Err(Mp3Error::HeaderError)
+    }
+
+    let version = match header[1] & 0b00011000u8 {
```

Comment:
Wouldn't it make harder to understand which bits we are checking?

Reply:
Well, I don't think so. Converting from hex to binary is really easy to do in your head, specially for small numbers like these, and you gain in readability because you need to reason about less symbols initially. This is just an opinion though, @est31 what do you think?

Reply:
I personally would have used hex numbers. But binary seems clear to me as well. Idk. Let's go with hex :).